lib/git-version-bump.rb

```diff
@@ -262,8 +267,16 @@ def self.caller_gemspec
 		# Grovel through all the loaded gems to try and find the gem
 		# that contains the caller's file.
 		Gem.loaded_specs.values.each do |spec|
-			search_dirs = spec.require_paths.map { |d| "#{spec.full_gem_path}/#{d}" } +
-			              [File.join(spec.full_gem_path, spec.bindir)]
+			# On Windows I have encountered gems that already have an absolute
+			# path, verify that the path is relative before appending to it
```

Comment:
How very extremely odd.  I wonder if that's an artifact of some gem, or a RubyGems behaviour.  Do you have a *nix box handy to check to see if the same gem+version has an absolute path in `spec.require_paths`?  If you don't, let me know an example gem+version and I can take a look.

Reply:
I believe this is an artifact of the installer I was using bringing precompiled gems in since the gems are most likely difficult to install on Windows. In this case I was using the Chef Workstation installer: https://downloads.chef.io/chef-workstation#windows which brings a full Windows ruby install. From there I installed and ran a gem which depends on `git-version-control` and got 2 different gems they build with absolute paths:

```ruby
=> Gem::Specification.new do |s|
  s.name = "ffi-yajl"
  s.version = Gem::Version.new("2.3.3")
  s.installed_by_version = Gem::Version.new("3.0.3")
  s.authors = ["Lamont Granquist"]
  s.date = Time.utc(2019, 12, 30)
  s.dependencies = [Gem::Dependency.new("rake", Gem::Requirement.new([">= 10.1"]), :development),
   Gem::Dependency.new("rspec", Gem::Requirement.new(["~> 3.0"]), :development),
   Gem::Dependency.new("pry", Gem::Requirement.new(["~> 0.9"]), :development),
   Gem::Dependency.new("rake-compiler", Gem::Requirement.new(["~> 1.0"]), :development),
   Gem::Dependency.new("rack", Gem::Requirement.new(["~> 2.0"]), :development),
   Gem::Dependency.new("libyajl2", Gem::Requirement.new(["~> 1.2"]), :runtime),
   Gem::Dependency.new("ffi", Gem::Requirement.new(["~> 1.5"]), :development)]
  s.description = "Ruby FFI wrapper around YAJL 2.x"
  s.email = "lamont@chef.io"
  s.executables = ["ffi-yajl-bench"]
  s.extensions = ["ext/ffi_yajl/ext/encoder/extconf.rb",
   "ext/ffi_yajl/ext/parser/extconf.rb",
   "ext/ffi_yajl/ext/dlopen/extconf.rb"]
  s.extra_rdoc_files = ["README.md", "LICENSE"]
  s.files = ["LICENSE",
   "README.md",
   "bin/ffi-yajl-bench",
   "ext/ffi_yajl/ext/dlopen/extconf.rb",
   "ext/ffi_yajl/ext/encoder/extconf.rb",
   "ext/ffi_yajl/ext/parser/extconf.rb"]
  s.homepage = "http://github.com/chef/ffi-yajl"
  s.licenses = ["MIT"]
  s.require_paths = ["C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.6.0/extensions/x64-mingw32/2.6.0/ffi-yajl-2.3.3",
   "lib"]
  s.required_ruby_version = Gem::Requirement.new([">= 2.2"])
  s.rubygems_version = "3.0.3"
  s.specification_version = 4
  s.summary = "Ruby FFI wrapper around YAJL 2.x"
  end
```

and 

```ruby
=> Gem::Specification.new do |s|
  s.name = "http_parser.rb"
  s.version = Gem::Version.new("0.6.0")
  s.installed_by_version = Gem::Version.new("3.0.3")
  s.authors = ["Marc-Andre Cournoyer", "Aman Gupta"]
  s.date = Time.utc(2013, 12, 11)
  s.dependencies = [Gem::Dependency.new("rake-compiler", Gem::Requirement.new([">= 0.7.9"]), :development),
   Gem::Dependency.new("rspec", Gem::Requirement.new([">= 2.0.1"]), :development),
   Gem::Dependency.new("json", Gem::Requirement.new([">= 1.4.6"]), :development),
   Gem::Dependency.new("benchmark_suite", Gem::Requirement.new([">= 0"]), :development),
   Gem::Dependency.new("ffi", Gem::Requirement.new([">= 0"]), :development),
   Gem::Dependency.new("yajl-ruby", Gem::Requirement.new([">= 0.8.1"]), :development)]
  s.description = "Ruby bindings to http://github.com/ry/http-parser and http://github.com/a2800276/http-parser.java"
  s.email = ["macournoyer@gmail.com", "aman@tmm1.net"]
  s.extensions = ["ext/ruby_http_parser/extconf.rb"]
  s.files = ["ext/ruby_http_parser/extconf.rb"]
  s.homepage = "http://github.com/tmm1/http_parser.rb"
  s.licenses = ["MIT"]
  s.require_paths = ["C:/opscode/chef-workstation/embedded/lib/ruby/gems/2.6.0/extensions/x64-mingw32/2.6.0/http_parser.rb-0.6.0",
   "lib"]
  s.rubygems_version = "3.0.3"
  s.specification_version = 4
  s.summary = "Simple callback-based HTTP request/response parser"
  end
```

Unfortunately they delete the gemspecs from the filesystem during their build to save space so I don't have the actual original gemspec to look at. Nothing in their build gemspec looks out of place: https://github.com/chef/ffi-yajl.